packages/service-library/src/servicelib/fastapi/long_running_tasks/_context_manager.py

```diff
@@ -20,14 +24,19 @@ class _ProgressManager:
     """
 
     def __init__(
-        self, update_callback: Optional[Callable[[str, float], Awaitable[None]]]
+        self,
+        update_callback: Optional[Callable[[str, float, TaskId], Awaitable[None]]],
```

Comment:
what are str and float? maybe have a meaningful alias

Reply:
and is the callable optional or message and progress?

Reply:
As explained the callable is optional as the user can choose not to receive these updates.
The message and progress are defined Optional since the `update` method can be called with just one of them.